packages/apps/plugins/plugin-space/src/components/SpacePresence.tsx

```diff
@@ -51,44 +46,55 @@ export const SpacePresence = () => {
-    );
+    return space.key.equals(viewer.spaceKey) && object.id === viewer.objectId && Date.now() - viewer.lastSeen < 30_000;
   });
 
-  return (
-    <div className='flex items-center'>
-      {intentPlugin && (
-        <Button variant='ghost' onClick={handleShare}>
-          <Users className={getSize(5)} />
-        </Button>
-      )}
-      <ObjectViewers viewers={viewers} />
-    </div>
-  );
+  return <ObjectPresence onShareClick={handleShare} viewers={viewers} />;
+};
+
+export type ObjectPresenceProps = {
+  size?: Size;
+  viewers?: ObjectViewer[];
+  showCount?: boolean;
+  onShareClick?: () => void;
 };
 
-const ObjectViewers: FC<{ viewers: ObjectViewer[] }> = ({ viewers }) => {
+export const ObjectPresence = (props: ObjectPresenceProps) => {
+  const {
+    onShareClick,
+    viewers = [],
+    size = 4,
+    showCount = !props?.size || (props.size !== 'px' && props.size > 4),
+  } = props;
   const { t } = useTranslation(SPACE_PLUGIN);
   return (
     <Tooltip.Root>
-      <Tooltip.Trigger className='flex items-center'>
-        <AvatarGroup.Root size={4} classNames='mie-5'>
-          <AvatarGroup.Label classNames='text-xs font-system-semibold'>{viewers.length}</AvatarGroup.Label>
-          {viewers.map((viewer) => (
-            <AvatarGroupItem.Root key={viewer.identityKey.toHex()}>
-              <Avatar.Frame>
-                {/* TODO(burdon): Why `href`? */}
-                <Avatar.Fallback href={viewer.identityKey.toHex()} />
-              </Avatar.Frame>
-            </AvatarGroupItem.Root>
-          ))}
-        </AvatarGroup.Root>
+      <Tooltip.Trigger className={mx('px-1 m-1 flex items-center')} onClick={onShareClick}>
+        {onShareClick && viewers.length === 0 && (
+          <DensityProvider density={'fine'}>
+            <Users className={mx(getSize(5), 'm-1')} />
+          </DensityProvider>
+        )}
+        {viewers.length > 0 && (
+          <AvatarGroup.Root size={size} classNames={'' + (size !== 'px' && size > 3 ? 'm-2 mie-4' : 'm-1 mie-2')}>
+            {viewers.length > 3 && showCount && (
+              <AvatarGroup.Label classNames='text-xs font-system-semibold'>{viewers.length}</AvatarGroup.Label>
+            )}
+            {viewers.slice(0, 3).map((viewer, i) => (
+              <AvatarGroupItem.Root
+                key={viewer.identityKey.toHex()}
+                color={getColorForValue({ value: viewer.identityKey.toHex(), type: 'color' })}
+              >
+                <Avatar.Frame style={{ zIndex: viewers.length - i }}>
+                  <Avatar.Fallback href={viewer.identityKey.toHex()} />
+                </Avatar.Frame>
+              </AvatarGroupItem.Root>
+            ))}
+          </AvatarGroup.Root>
+        )}
       </Tooltip.Trigger>
       <Tooltip.Content collisionPadding={4}>
-        <span>{t('presence label')}</span>
+        <span>{viewers.length > 0 ? viewers.length + ' ' + t('presence label') : t('share space')}</span>
```

Comment:
This should use the `t('key', {count: viewers.length})` syntax instead of hardcoding a format itself.
https://www.i18next.com/translation-function/plurals